app/component/search/SearchMainContainer.js

```diff
@@ -0,0 +1,212 @@
+import React from 'react';
+import { intlShape } from 'react-intl';
+import Tab from 'material-ui/Tabs/Tab';
+
+import { setEndpoint, setUseCurrent } from '../../action/EndpointActions';
+import { executeSearch } from '../../action/SearchActions';
+import FakeSearchBar from './FakeSearchBar';
+import FakeSearchWithButton from './FakeSearchWithButton';
+import GeolocationOrInput from './GeolocationOrInput';
+import SearchModal from './SearchModal';
+
+class SearchMainContainer extends React.Component {
+  static contextTypes = {
+    executeAction: React.PropTypes.func.isRequired,
+    getStore: React.PropTypes.func.isRequired,
+    router: React.PropTypes.object.isRequired,
+    intl: intlShape.isRequired,
+  };
+
+  constructor(args) {
+    super(...args);
+    this.searchInputs = [];
```

Comment:
eslint wants that initializer after lifetime methods, and seemed better to have properties up.